snooty/types.py

```diff
@@ -311,6 +312,21 @@ def render_constants(self) -> Tuple["ProjectConfig", List[Diagnostic]]:
 
     def read(self, path: Path) -> Tuple[str, List[Diagnostic]]:
         text = path.read_text(encoding="utf-8")
+        PAT_GIT_MARKER = re.compile(r"^<<<<<<< .*?^=======\n.*?^>>>>>>>", re.M | re.S)
+        match = PAT_GIT_MARKER.search(text)
+
+        if match:
+             diagnostics = []
+             lineno = text.count("\n", 0, match.start())
+             diagnostics.append(
+                Diagnostic.error(
+                    "merge conflict artifact found in source documentation", lineno
+                )
+             )
+             
+             return (text, diagnostics)
```

Comment:
We should still run substitutions to try and pick up as many diagnostics as possible. So either way you're going to be running `self.substitute(text)`: it's just that you'll be merging the diagnostic list